docs/topics/articles.md

```diff
@@ -0,0 +1,184 @@
+# `Article`s and `ListingPage`s
+
+`Article`s are publishable content types that live in collections. Normally
+articles are mounted under a parent's URL.
+
+An example of an article is a Press Release, which would be mounted under a
+Press Release Listing Page. The Listing Page has the URL "/press" and thus all
+the Press Releases have the URL "/press/<slug>".
+
+## `ArticleBase`
+
+The `icekit.articles.models.ArticleBase` inherits `title`, `slug` and
+publishing fields, and each `ArticleBase` subclass implements some key
+functionality:
+
+1. **Links to parents.**
+In order for an Article to know its URL (which is based on a parent's URL), it
+should define a `parent` link, which is normally a link to a `ListingPage`.
```

Comment:
The reason for coupling an article to a ListingPage is so that they inherit the Page's URL and that there is only one URL per article, and that can be decided by editors. So in some sites, press releases will be at /press-releases/<slug>. In others, they'll at /media/releases/<slug>.

In some cases we're intending to link some articles to one listing page and some to another. SFMOMA's watch/read/listen articles are a case in point. In other cases, we're not. e.g. all press releases belong under press. In this case, I've defined `@property parent` to return the first draft press release listing page.

Other pages may (and do) list articles in different ways, but this arrangement is the one that defines the URL.

Agreed that the on_delete behaviour needs fixing, and documentation needs clarifying, and possibly ArticleBase should become ContentCollection or something.

The reason Articles are not page_types in the page tree is that a) they have different admin_listing requirements (a tree is not a useful way to view/sort Articles) b) the page tree gets very unwieldy with sometimes thousands of items to sort and filter c) we don't want Articles to be children of arbitrary pages, only a defined subset.